### DISCUSSION TITLE
[@types/chrome-apps] has incorrect return type for webview.ClearData() function

### DISCUSSION THREAD

**Discussion by bswanwick** [DefinitelyTyped/DefinitelyTyped#69283] [Issues with a @types package]
I'm not sure if I should report this as an issue or just make a PR to fix it, but the type doesn't match the return value at runtime.  I have an NW.js application (which is running Chromium 120).  I am calling `clearData` function on a `<webview>` tag and the return value from clearData function is actually a boolean, though the type definition says it is a void.  

Link to chrome docs doesn't explain the return type:  https://developer.chrome.com/docs/apps/reference/webviewTag#method-clearData

But at runtime the function definitely returns a boolean value, not void.

[a comment by an automated account is not shown]

**Comment by pyle** [comment]
I can do a little more digging next week, these are the chromium typings we would have derived this from - https://source.chromium.org/chromium/chromium/src/+/main:tools/typescript/definitions/webview_tag.d.ts;l=300;drc=99ef4a255672675b6593a7c0c1ca079de3927f9f

Will need to figure out why there is a discrepancy from typings and runtime.

**Reply by pyle** [reply]
Had a poke at this, i'm not a chromium engineer but have had some experience in the codebase before.
What i believe is the code for this feature, `clearData` , does return a boolean, source[^1] (this is based off of the original bug[^2]).

The reason the [docs](https://developer.chrome.com/docs/apps/reference/webviewTag#method-clearData) show `Promise<void>` is because api definition listed [here](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/chrome/common/extensions/api/webview_tag.json#815) doesn't provide an explicit `return` property (likely an oversight when the feature was developed almost ~9 years ago?[^2])


Based on the source[^1] you're probably right to ignore the return value, it'll only return false if the underlying storage cannot be reached (it's likely you'll have other issues before this happens).

We should do the following:
1. [Raise a chromium/chrome](https://issues.chromium.org/new) bug to:
  a. Update the docs listed, and 
  b. see if the above is correct - https://developer.chrome.com/docs/apps/reference/webviewTag#method-clearData
3. Then once we have confirmation the API docs will likely get re-posted and at that point feel free to send a PR for the typing


[^1]: https://chromium.googlesource.com/chromium/src/+/fde08e6daf69acc4143cc6f10cf7e61199f6a511/extensions/browser/guest_view/web_view/web_view_guest.cc#746
[^2]: https://issues.chromium.org/issues/41127533#comment21 

---

Original commit - https://chromium.googlesource.com/chromium/src/+/cbff2a79e3aafa333159dbc467efcb8a5dc124f7

**Comment by bswanwick** [comment]
I just tested again in Chromium v123 (latest available to nw.js users) and the result is the same.
